### ISSUE TITLE
Self-Rewarding Language Models

### ISSUE THREAD

**Issue by ooooohira** [ooooohira/paper_summary#14]
### 論文リンク
[https://arxiv.org/abs/2401.10020](https://arxiv.org/abs/2401.10020)

### 著者
Weizhe Yuan, Richard Yuanzhe Pang, Kyunghyun Cho, Xian Li, Sainbayar Sukhbaatar, Jing Xu, Jason Weston

### 投稿年
Thu, 18 Jan 2024

## 新規性や利点
Open Assistant からの少数のシードデータから開始して、Claude 2、Gemini Pro、GPT4などの、大量の教師ありデータから学習されたモデルを凌駕(ただしGPT-4 Turboには負けている)。

## 提案手法
2.1 Initialization
Llama 2 70B をITF(promptとそれに対する回答)とEFTデータ(評価のためのpromptとそれに対する回答、ただし順位付き)を用意して、fine-tuningする。
Open Assistant　＝　ITF + EFT(4830個のexample)

2.2 Self-Instruction Creation
Wang et al. [2023] and Honovich et al. [2023]の手法を用いてITFデータを用いてprompt作成(このときは学習対象のモデルではなく固定されたLlama 2-Chat 70B を用いた)→得たプロンプトからの回答をN個作成→それぞれに対してLLM-as-a-Judge を適応して5段階評価。

2.3 Instruction Following Training
2.3で得たN個の5段階評価済みのものからmax,min評価のものをペアとして格納(AI Feedback Training dataと呼ばれる)し、DPOを走らせる。

全体のアルゴリズム
<img width="859" alt="スクリーンショット 2024-04-26 20 10 06" src="https://github.com/ooooohira/paper_summary/assets/131585401/12753429-3073-47cf-a529-8a9a7d834b8c">
おそらく、AIFT(Mi)はMiのみが作ったAI Feedback Training dataだと思う。
M1の時、EFTではどのようにfine-tuningしているのか？→追記：SFT baselineを用いて5点満点の評価してもらって、人間の評価と一致していたら、それをEFTに含める。

## 実験
色々なタスクでイテレーションを回して性能がどれくらい改善するのかを、Instruction FollowingとReward Modelingの観点で調べた。

評価方法：
Instruction Following
いろんなタスクや評価指標で確認
baselineはITFでfine-tuningしたものを用いた。

Reward Modeling
3つの順位相関係数で人間の順位と比較した。
ここの評価では、ETF作成時にtrainingに使用しなかった531個のevaluationデータを用いた。

## 結果と考察
<img width="1095" alt="スクリーンショット 2024-04-26 23 23 04" src="https://github.com/ooooohira/paper_summary/assets/131585401/ee22c754-47ba-4800-841f-6d8234395557">
<img width="778" alt="スクリーンショット 2024-04-26 23 23 45" src="https://github.com/ooooohira/paper_summary/assets/131585401/a1baba44-af7a-413e-ac81-79b6f73fda45">
<img width="784" alt="スクリーンショット 2024-04-26 23 24 32" src="https://github.com/ooooohira/paper_summary/assets/131585401/7d5c6e7d-431d-462f-a5fe-e3567cc17b02">
基本的にほとんどの場合で、イテレーションが進むにつれて性能が改善(人手評価も含めて)
しかしNLP タスク、数学とかロジカル的なタスクの性能は上がりずらく場合によってはちょっと下がっている。→Open Assistant からのシードプロンプトが論理推論タスクや、NLPタスクを軽視する傾向にあるため。

<img width="1202" alt="スクリーンショット 2024-04-26 23 26 49" src="https://github.com/ooooohira/paper_summary/assets/131585401/d9471641-4538-4fbb-b168-fb982a4ffd24">
報酬モデルとしての性能(人間との相関)も報酬モデルとしてのタスクと明示的には関係ないDPOを使っているのにも関わらず、イテレーションごとに上がっている(少しだけど)。
ETFデータでのSFTが報酬モデルとしての性能を大きく上昇させることが分かる。

報酬モデルに使うpromptの種類で性能が大きく変わる。

## 感想や疑問点
Data distribution analysisで重なっていると望ましいと書いてあるけど、どういうこと？？
なんで報酬モデルとしての性能は上がるのに、NLP タスク、数学とかロジカル的なタスクの性能は下がったりすることがあるのだろうか(どちらのタスクもOpen Assistant からのシードプロンプトは軽視しているはずなのに)？？→ほぼ誤差のようなものかな？

## 参考
